source/common/http/http1/codec_impl.cc

```diff
@@ -477,6 +477,7 @@ void ConnectionImpl::onMessageCompleteBase() {
 
 void ConnectionImpl::onMessageBeginBase() {
   ENVOY_CONN_LOG(trace, "message begin", connection_);
+  protocol_ = Protocol::Http11;
```

Comment:
Can you add a small comment here on what this case is covering? (Basically the client switching between 1.0 and 1.1)